lib/rouge/guessers/disambiguation.rb

```diff
@@ -86,7 +86,11 @@ def match?(filename)
         next Mathematica if contains?('(*')
         next Mathematica if contains?(':=')
 
+        next Mason if matches?(/<%(def|method|text|doc|args|flags|attr|init|once|shared|perl|cleanup|filter)([^>]*)(>)/)
+
         next Matlab if matches?(/^\s*?%/)
+
+        next Mason if matches?(/(<\/?%|<&)/)
```

Comment:
To improve readability:

```suggestion
        next Mason if matches? %r!(</?%|<&)!
```

Reply:
Applied, thanks.